### PR TITLE
Resolve common static analysis issues

### PR DESCRIPTION
**Pull request by timtebeek**
As discussed with @jeremyg484, these are some automated code changes according to a list of [common static analysis issues](https://docs.openrewrite.org/recipes/staticanalysis/commonstaticanalysis). The idea is to resolve such issues, with the option to recreate a smaller or larger set of changes easily.

I've also done a dry run of this same recipe set against all of the Micronaut projects we currently support, which resulted in these statistics in terms of issues that would resolve.
image (8)
For instance using the diamond operator is a common one that should not be controversial, and could be applied in a couple minutes.

I'll only open this one to gather feedback; no worries if it's not to your taste, these changes were easy to make. Curious to hear your thoughts!

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/logging/Log4j2.java
Comment (alvarosanchez) [inline starter-core/src/main/java/io/micronaut/starter/feature/logging/Log4j2.java]: This one was probably missing `static` mistakenly, and then it wouldn't be renamed

**Review by alvarosanchez — CHANGES_REQUESTED** [review]
@timtebeek thank you very much! Overall LGTM, but there is one change that should add static instead of renaming.

This repo is probably the worst in the sense that it is likely the only one we have that doesn't enforce Checkstyle and doesn't run Sonar analysis.

But I do think this is going to be beneficial for all of our repos.

File: starter-core/src/main/java/io/micronaut/starter/feature/logging/Log4j2.java
Comment (timtebeek) [reply starter-core/src/main/java/io/micronaut/starter/feature/logging/Log4j2.java]: You're right! Fixed just now & good that you spotted this. :)

**Review by alvarosanchez — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]

**Comment by sdelamo** [comment]
@timtebeek thanks for the PR. Please, feel free to create a PR for other repositories.

**Comment by timtebeek** [comment]
Thanks both! I've gone ahead and logged two more in this repository, such that we don't have to wait for CI approval first.

- [x] https://github.com/micronaut-projects/micronaut-starter/pull/2002
- [x] https://github.com/micronaut-projects/micronaut-starter/pull/2003

I've added the same statistics as seen above in each PR, such that you get a sense of if it would make sense to run those recipes across more repositories. I've held off on creating more PRs as to not overwhelm you with review work, and such that you have a say in what PRs get created, especially when there's relatively few changes.

If there's any repository in particular you'd want me to target with new PRs let me know; it's pretty trivial through the platform, as seen through the links at the start of each PR. And you're also welcome to explore yourselves of course.

**Comment by timtebeek** [comment]
These are the [stats for Micronaut Core 4.1.x](https://app.moderne.io/results/yBErCyx9g)
```
    114 "org.openrewrite.staticanalysis.ExplicitInitialization"
     86 "org.openrewrite.staticanalysis.UseDiamondOperator"
     63 "org.openrewrite.staticanalysis.EqualsAvoidsNull"
     55 "org.openrewrite.staticanalysis.UnnecessaryParentheses"
     36 "org.openrewrite.staticanalysis.FinalizePrivateFields"
     19 "org.openrewrite.staticanalysis.ReplaceLambdaWithMethodReference"
     15 "org.openrewrite.staticanalysis.NeedBraces"
     13 "org.openrewrite.staticanalysis.FinalClass"
      8 "org.openrewrite.staticanalysis.ReplaceStringBuilderWithString"
      6 "org.openrewrite.staticanalysis.LowercasePackage"
      5 "org.openrewrite.staticanalysis.MultipleVariableDeclarations"
      5 "org.openrewrite.staticanalysis.InlineVariable"
      4 "org.openrewrite.staticanalysis.RenamePrivateFieldsToCamelCase"
      4 "org.openrewrite.staticanalysis.NoEmptyCollectionWithRawType"
      3 "org.openrewrite.staticanalysis.LambdaBlockToExpression"
      3 "org.openrewrite.staticanalysis.IsEmptyCallOnCollections"
      2 "org.openrewrite.staticanalysis.UnnecessaryExplicitTypeArguments"
      2 "org.openrewrite.staticanalysis.SimplifyBooleanReturn"
      2 "org.openrewrite.staticanalysis.ModifierOrder"
      2 "org.openrewrite.staticanalysis.MinimumSwitchCases"
      1 "org.openrewrite.staticanalysis.RenameLocalVariablesToCamelCase"
      1 "org.openrewrite.staticanalysis.RemoveExtraSemicolons"
      1 "org.openrewrite.staticanalysis.NoValueOfOnStringType"
      1 "org.openrewrite.staticanalysis.NoRedundantJumpStatements"
      1 "org.openrewrite.staticanalysis.MethodNameCasing"
      1 "org.openrewrite.staticanalysis.FallThrough"
```
I wouldn't recommend just blindly running all of them together, but perhaps it makes sense to create a pr that combines [`UseDiamondOperator`](https://docs.openrewrite.org/recipes/staticanalysis/usediamondoperator) & [`EqualsAvoidsNull`](https://docs.openrewrite.org/recipes/staticanalysis/equalsavoidsnull) together for some quick common fixes in micronaut-core.

**Pull request merged**
